### PR TITLE
Update infrastructure walkthrough to use Azure Voting App

### LINKED ISSUE #1132: As a developer, I want the Azure Single cluster with Keyvault walkthrough referenced from the docs page
**As a:**
Developer
**I want:**
Single Cluster with Keyvault walkthrough referenced from the docs main page 
**So that:**
I can find the walkthrough
**Describe the solution you'd like:**
Update the following page https://github.com/microsoft/bedrock/tree/docs_spk
**Acceptance Criteria:**
Page updated with correct walkthrough link
**Describe alternatives you've considered:**
None
**Additional context:**
None
**Does this require updates to documentation?:**

### PR DESCRIPTION
**Pull request by timfpark**
Closes #1132

### CODE REVIEW

**Comment by jred55** [comment]
Adding single keyvault walkthrough

File: docs/singleKeyvault/README.md [hunk 1]
```diff
@@ -0,0 +1,367 @@
+# Deploying Single Cluster with Keyvault 
+
+If you followed our first workload walkthrough you saw how Bedrock enables you to scaffold and generate Terraform deployment scripts. In the first workload walkthrough, to demonstrate simple GitOps workflow, we deployed a simple cluster using Azure Simple Terraform template. However in the real world, in order to have secure deployments, we need a way to store the essential secrets in Keyvault. In addition, we will also deploy a vnet to provide isolation to our cloud infrastructure. 
+
+In upcoming advanced challenges, we will be using the Bedrock automation to repeat the cluster creation process by scaffolding configurations to deploy multiple clusters. Since all these clusters use common resources like Keyvault, Storage Account and a Vnet, we will deploy these resources using azure-common-infra template. The environment provisioned using this template is a dependency for other environments (azure-single-keyvualt) we will be using in the subsequent walkthroughs.
```
Comment (timfpark) [inline docs/singleKeyvault/README.md]: Consider not using word "challenges" but perhaps scenario?

File: README.md [hunk 2]
```diff
@@ -13,9 +13,8 @@ Bedrock helps you:
 * [Walkthrough: Deploying a First Workload](./docs/firstWorkload)
 * [Deep Dive: Why GitOps?](./docs/why-gitops.md)
 
-## Infrastructure Management
-* [Walkthrough: Deploying and Maintaining Infrastructure](./docs/infrastructure.md)
-* [CLI Reference](https://github.com/CatalystCode/spk/blob/master/guides/cloud-infra-management.md)
+## First Cluster
```
Comment (timfpark) [inline README.md]: I would recommend just keeping the two cluster pieces together.  The day two part I'll just cover in the deep dive section.

File: docs/singleKeyvault/README.md [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+variables:
+  address_space: '10.39.0.0/16'
+  keyvault_name: 'mykeyvault'
+  global_resource_group_name: 'my-global-rg'
+  service_principal_id: '91896545-0aa8-4444-5555-111461be44a6'
+  subnet_name: 'mysubnet'
+  subnet_prefix: '10.39.0.0/24'
+  vnet_name: 'myvnet'
+```
+Now that we have these variables filled in, we will use 'spk generate' command to generate terraform tfvars file that we will use to provision the commond infrastructure. Navigate to azure-common-infra/westus folder and run the following command
```
Comment (timfpark) [inline docs/singleKeyvault/README.md]: typo: commond -> command

File: docs/singleKeyvault/README.md [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+$ az group create -l westus2 -n my-cluster-rg
+```
+Next, to scaffold infrastructure, we will use spk scaffold command at the root level and the cluster level
+
+At the same level as your azure-common-infra directory, run the following command
+
+```
+spk infra scaffold --name azure-single-keyvault --source https://github.com/microsoft/bedrock --version master --template cluster/environments/azure-single-keyvault
+```
+This creates a directory named azure-single-keyvault and places global defintion.yaml inside the directory. Now, nagigate to this directory and create cluster specific configurations by running the following commnad
```
Comment (timfpark) [inline docs/singleKeyvault/README.md]: nagigate -> navigate

File: docs/singleKeyvault/README.md [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
+  subnet_name: <insert value>
+  network_plugin: azure
+  network_policy: azure
+  oms_agent_enabled: 'false'
+  enable_acr: 'false'
+  acr_name: <insert value>
+```
+
+Next we'll fill all of the empty items in this template with config values.
+Note: Use `storage_account_name`, `access_key` , `container_name`, `keyvault_name`, `keyvault_resource_group` and `vnet_name` from previous "Deploying commond infrastrucute" step.
```
Comment (timfpark) [inline docs/singleKeyvault/README.md]: typo: infrastrucute -> infrastructure

File: docs/singleKeyvault/README.md [hunk 6]
```diff
@@ -0,0 +1,367 @@
...
+```
+
+```yaml
+name: cluster
+source: 'https://github.com/microsoft/bedrock'
+template: cluster/environments/azure-single-keyvault
+version: master
+backend:
+  storage_account_name: 'mystorageaccount'
+  access_key: 'CENp3G0qvo4jB1HduRO10ga0jNrN+b4gMibuAp63qZBDRNzXYZrP7dUu8XM4lca6HL4RobXCfhjvAAAAAbbbb=='
```
Comment (timfpark) [inline docs/singleKeyvault/README.md]: is this a fake access_key?

File: docs/singleKeyvault/README.md [hunk 7]
```diff
@@ -0,0 +1,367 @@
...
+  dns_prefix: 'spk'
+  flux_recreate: 'false'
+  gitops_ssh_url: 'git@ssh.dev.azure.com:v3/myorg/app-cluster-manifests'
+  gitops_path: 'prod'
+  gitops_ssh_key: '~/cluster-deployment/keys'
+  gitops_url_branch: master
+  keyvault_name: 'mykeyvault'
+  keyvault_resource_group: 'my-global-rg'
+  resource_group_name: 'my-cluster-rg'
+  ssh_public_key: 'ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDUcBzqjBc59Ypa+Y2Cc9z+wDldZnSEGoJt+sUbux/KrczQmHmKqpdW50zMSY4MYhdfJsn902mZad4qOVj/KvQwwl7cGyWqzK+yEw/CrgqCX9wzloJrq75M1V3Qaaaaaaaaaaaaabbbbbbb91lnmtzGyOIJJlHoxm4TPR8tRhWeAcb6mRBKOeGQSSNekyi08dtYhYHlWFXaSZzqVevgiNYCGkcgXbPE1fE6Da2SAmOdBwANCHE8OZXnh yourname@org.com'
```
Comment (timfpark) [inline docs/singleKeyvault/README.md]: is this a faked ssh key?

Comment (jred55) [reply docs/singleKeyvault/README.md]: Yes

File: docs/singleKeyvault/README.md [hunk 8]
```diff
@@ -0,0 +1,367 @@
+# Deploying Single Cluster with Keyvault 
+
+If you followed our first workload walkthrough you saw how Bedrock enables you to scaffold and generate Terraform deployment scripts. In the first workload walkthrough, to demonstrate simple GitOps workflow, we deployed a simple cluster using Azure Simple Terraform template. However in the real world, in order to have secure deployments, we need a way to store the essential secrets in Keyvault. In addition, we will also deploy a vnet to provide isolation to our cloud infrastructure.
```
Comment (edaena) [inline docs/singleKeyvault/README.md]: to demonstrate "a" simple GitOps workflow -> missing the 'a'

File: docs/singleKeyvault/README.md [hunk 9]
```diff
@@ -0,0 +1,367 @@
...
+If you followed our first workload walkthrough you saw how Bedrock enables you to scaffold and generate Terraform deployment scripts. In the first workload walkthrough, to demonstrate simple GitOps workflow, we deployed a simple cluster using Azure Simple Terraform template. However in the real world, in order to have secure deployments, we need a way to store the essential secrets in Keyvault. In addition, we will also deploy a vnet to provide isolation to our cloud infrastructure. 
+
+In upcoming advanced challenges, we will be using the Bedrock automation to repeat the cluster creation process by scaffolding configurations to deploy multiple clusters. Since all these clusters use common resources like Keyvault, Storage Account and a Vnet, we will deploy these resources using azure-common-infra template. The environment provisioned using this template is a dependency for other environments (azure-single-keyvualt) we will be using in the subsequent walkthroughs.
+
+## Deplying the common infrastructure:
+
+Before attempting to deploy the infrastructure environments, you will also need to create an Azure Storage Account. You can do this in Azure Portal, or by using the Azure CLI:
```
Comment (edaena) [inline docs/singleKeyvault/README.md]: nit: removed a few workds
"To deploy the infrastructure environments, you will need to..."

File: docs/singleKeyvault/README.md [hunk 10]
```diff
@@ -0,0 +1,367 @@
...
+
+To create a resource group, you can use the following command 
+
+```
+$ az group create -l westus2 -n my-global-rg
+```
+
+### Create Storage Account in Azure
+
+Before attempting to deploy the infrastructure environments, you will also need to create an Azure Storage Account. You can do this in Azure Portal, or by using the Azure CLI:
```
Comment (edaena) [inline docs/singleKeyvault/README.md]: nit: removed words
To deploy infrastructure environments, you will need to create an Azure Storage Account.

File: docs/singleKeyvault/README.md [hunk 11]
```diff
@@ -0,0 +1,367 @@
...
+
+In upcoming advanced challenges, we will be using the Bedrock automation to repeat the cluster creation process by scaffolding configurations to deploy multiple clusters. Since all these clusters use common resources like Keyvault, Storage Account and a Vnet, we will deploy these resources using azure-common-infra template. The environment provisioned using this template is a dependency for other environments (azure-single-keyvualt) we will be using in the subsequent walkthroughs.
+
+## Deplying the common infrastructure:
+
+Before attempting to deploy the infrastructure environments, you will also need to create an Azure Storage Account. You can do this in Azure Portal, or by using the Azure CLI:
+
+### Resource Group Requirement
+
+This environment requires a single resource group be created.  The requisite variable is `global_resource_group_name`.  To use the Azure CLI to create the resource group, see [here](https://github.com/microsoft/bedrock/blob/master/cluster/azure/README.md).
```
Comment (edaena) [inline docs/singleKeyvault/README.md]: This environment requires a resource group. To create one with the Azure CLI, see here.

File: docs/singleKeyvault/README.md [hunk 12]
```diff
@@ -0,0 +1,367 @@
...
+  global_resource_group_name: <insert value>
+  service_principal_id: <insert value>
+  subnet_name: <insert value>
+  subnet_prefix: <insert value>
+  vnet_name: <insert value>
+```
+`scaffold` has downloaded the template locally, extracted all of the variables for the template, and provided defaults where possible for all of the variables.
+
+Let's fill in the variables for common-infra infrastructure variables. 
+
```
Comment (edaena) [inline docs/singleKeyvault/README.md]: Note: `global_resource_group_name` is the resource group that was created in [Resource Group Requirement](#resource-group-requirement).

File: docs/singleKeyvault/README.md (hunk 6 above)
Comment (jred55) [reply docs/singleKeyvault/README.md]: yes

File: docs/singleKeyvault/README.md (hunk 11 above)
Comment (jred55) [reply docs/singleKeyvault/README.md]: Made a slight change

**Pull request merged**